### ISSUE TITLE
client.debug

### ISSUE THREAD

**Issue by till** [wolfgang-werner/lexoffice-go-api#1]
Hey, 

just trying out your code and noticed that `client.debug` doesn't work as it's not exported (capitalized). Generally, maybe an option pattern would work nicer to setup the client?

```go
lexoffice.NewClient(apiKey,
  lexoffice.WithDebug(true),
  lexoffice.WithHttpClient(myOwnClient),
)

func NewClient(apiKey, ...opts) (*lexoffice.Client, error) {
    // ...
    // probably range over the opts
}
```

Thoughts?

**Comment by wolfgang-werner** [comment]
This is a good point. I initially created the library as one of my first go attempts and did not plan to externalize the debug (only used in in my tests). But of course your suggestion is much better. As I'm planning to use the lexoffice integration much more in the future, even extending it to more parts of the public API, may I ask if you are interested in bringing this forward? 

I'm completely open to any suggestions, pr's etc. and can work on it again. Thoughts?

Just saw that there is another project for the topic you are engaged with. Good to know!

**Comment by till** [comment]
@wolfgang-werner 👋 Sorry I closed this. Had no expected a response.

I started working on/fixing up another lib called golexoffice as it seemed to support more calls originally.

When I asked its author, they archived it and told me I could maintain it going forward as they are no longer using it.

Generally, we need contact management and invoices so far. I am also happy to collaborate on it going forward.

What I did so far is here:
https://github.com/hostwithquantum/golexoffice

Mostly added error handling, and a basic test suite. The library was also really procedural.
